rep-0146.rst

```diff
@@ -0,0 +1,80 @@
+REP: 146
+Title: ROS Robot Parameter Naming Convention
+Author: Kei Okada <k-okada@jsk.t.u-tokyo.ac.jp>
+Status: Draft
+Type: Informational
+Content-Type: text/x-rst
+Created: 31-Jly-2015
+Post-History:
+
+Abstract
+========
+
+This REP aims to define the best practices for robot related parameter
+name space for making robot-agnostic packages.
+
+
+Motivation
+==========
+
+As the number of ROS-supported robot increases, we will write
+robot-agnostic packages, but still they need robot-specific
+information as calibration parameters.
+
+A `robot_description` parameter is widely shared naming convention
+among ROS community and this REP proposes such conventions for robot specific information on
+parameter, so application program can build their program on top of that.
+
+
+Specifications
+==============
+
+Parameters
+----------
+
+robot_description
+'''''''''''''''''
+
+`robot_description` stores robot kinematics / geometry information using
+URDF format.
+
+
+robot/type
+''''''''''
+
+`robot/type` stores robot type, for example 'pr2' in PR2 robot case [1]
+
+
+robot/name
```

Comment:
I think that the namespacing here doesn't really make sense. To be able to introspect/discover this field a longer /more unique name would be better. I'd suggest `robot_name` and `robot_type`.

This also gets rid of the ambiguity of what namespace it applies to. The keys will be in the root of the namespace of the robot. So if you're searching for multiple robots, you don't have to look for a `name` field, and then traverse upward looking for a namespace `robot` and then traverse upward once more to find the `robot_description`

Reply:
thanks for review, although it will differ from original PR2's name http://github.com/pr2-debs/pr2-core/blob/master/root/etc/ros/robot.yaml, but it much make sense